src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -339,8 +357,31 @@
        :on-done (fn [{:keys [success? get-parsed-response]}]
                   (if success?
                     (do (swap! state assoc :is-leo-whitelisted? true)
-                        (this :-get-clusters-list-if-whitelisted))
+                        (this :-get-clusters-list-if-whitelisted)
+                        (this :-schedule-cookie-refresh-if-whitelisted))
                     (swap! state assoc :server-response {:server-error (get-parsed-response false)})))}))
+
+   :-schedule-cookie-refresh-if-whitelisted
+   (fn [{:keys [props state locals this]}]
+     (let [{{:keys [clusters]} :server-response} @state]
+       (when (and (not (:dead? @locals)) (:is-leo-whitelisted? @state))
+         (do (when (contains-statuses clusters ["Running"]) (this :-process-running-clusters))
+             (js/setTimeout #(this :-schedule-cookie-refresh-if-whitelisted) 120000)))))
+
+   :-process-running-clusters
+   (fn [{:keys [props state locals this]}]
+     (let [{{:keys [clusters]} :server-response} @state
+           running-clusters (filter #(= "Running" (:status %)) clusters)]
```

Comment:
Another "let's have fun with syntax" option that you should feel free to ignore:
```cljs
(filter (partial = "Running") (map :status clusters))
```